modules/core/include/opencv2/core.hpp

```diff
@@ -1118,6 +1118,13 @@ CV_EXPORTS_W void flip(InputArray src, OutputArray dst, int flipCode);
  */
 CV_EXPORTS_W void flipND(InputArray src, OutputArray dst, int axis);
 
+/** @brief Broadcast the given Mat to the given shape.
+ * @param src input array
+ * @param dst output array that has the given shape
+ * @param shape target shape
+ */
+CV_EXPORTS_W void broadcastTo(InputArray src, const std::vector<int>& shape, OutputArray dst);
```

Comment:
```
const std::vector<int>& shape
```

Why does `InputArray` not work here? (at least it could support `std::array<>` too).

Reply:
1. I followed what `transposeND` does:

https://github.com/opencv/opencv/blob/3aeaa3402389fc55e53fbc6f5741ca29f51032ee/modules/core/include/opencv2/core.hpp#L1758-L1766

2. `shape` should be a list of intergers and the signature `const std::vector<int>&` helps those who read the document to be aware of that.

3. I would say using `std::array` is minor case. `const std::vector<int>&` satisfies most usages.

Reply:
Decided to use `InputArray` in a discussion with @vpisarev .